### PR TITLE
Fixes from PHPStan level 0

### PR DESCRIPTION
**Pull request by carusogabriel**
Some fixes found by [`PHPStan`](https://github.com/phpstan/phpstan) level 0

### CODE REVIEW

File: src/ChangeStream.php [hunk 1]
```diff
@@ -143,8 +143,8 @@ private function extractResumeToken($document)
         if ($document === null) {
             throw new ResumeTokenException("Cannot extract a resumeToken from an empty document");
         }
-        if ($document instanceof Serializable) {
-            return $this->extractResumeToken($document->bsonSerialize());
+        if ($document instanceof \Serializable) {
+            $this->extractResumeToken($document->bsonSerialize());
```
Comment (jmikola) [inline src/ChangeStream.php]: @carusogabriel: This looks incorrect. Is PHPStan complaining that a `void` function is returning the result of another `void` function? If so, I think the correct fix is to add a solitary `return;` on the line below. The point of this is to recurse on the result of `$document->bsonSerialize()`, while this change allows execution of the current function to continue.

File: src/ChangeStream.php [hunk 2]
```diff
@@ -143,8 +143,8 @@ private function extractResumeToken($document)
         if ($document === null) {
             throw new ResumeTokenException("Cannot extract a resumeToken from an empty document");
         }
-        if ($document instanceof Serializable) {
-            return $this->extractResumeToken($document->bsonSerialize());
+        if ($document instanceof \Serializable) {
```
Comment (jmikola) [inline src/ChangeStream.php]: I believe this should have been left as-is, while adding `use MongoDB\BSON\Serializable;` above.

File: src/GridFS/Bucket.php [hunk 3]
```diff
@@ -17,6 +17,7 @@
 
 namespace MongoDB\GridFS;
 
+use MongoDB\Collection;
```
Comment (jmikola) [inline src/GridFS/Bucket.php]: It looks like this was needed for the `getChunksCollection` return type. Good catch.

File: src/GridFS/ReadableStream.php [hunk 4]
```diff
@@ -175,7 +175,7 @@ public function readBytes($length)
     public function seek($offset)
     {
         if ($offset < 0 || $offset > $this->file->length) {
-            throw new InvalidArgumentException(sprintf('$offset must be >= 0 and <= %d; given: %d', $length, $offset));
+            throw new InvalidArgumentException(sprintf('$offset must be >= 0 and <= %d; given: %d', $this->file->length, $offset));
```
Comment (jmikola) [inline src/GridFS/ReadableStream.php]: https://github.com/mongodb/mongo-php-library/pull/461 adds a regression test for this.

File: src/GridFS/WritableStream.php [hunk 5]
```diff
@@ -141,7 +141,7 @@ public function close()
     /**
      * Return the stream's file document.
      *
-     * @return stdClass
+     * @return \stdClass
```
Comment (jmikola) [inline src/GridFS/WritableStream.php]: I'd prefer to add a `use stdClass` up top for this. Will do so when merging.

File: src/MapReduceResult.php [hunk 6]
```diff
@@ -77,7 +77,7 @@ public function getExecutionTimeMS()
      * Return the mapReduce results as a Traversable.
      *
      * @see http://php.net/iteratoraggregate.getiterator
-     * @return Traversable
+     * @return \Traversable
```
Comment (jmikola) [inline src/MapReduceResult.php]: Same here re: `use Traversable` up top.

File: src/Operation/ChangeStream.php [hunk 7]
```diff
@@ -165,7 +165,7 @@ private function createChangeStreamOptions()
     /**
      * Create the aggregate pipeline with the changeStream command.
      *
-     * @return Command
+     * @return \MongoDB\Driver\Command
```
Comment (jmikola) [inline src/Operation/ChangeStream.php]: Will add `use MongoDB\Driver\Command;` up top.

File: src/Operation/MapReduce.php [hunk 8]
```diff
@@ -17,6 +17,7 @@
 
 namespace MongoDB\Operation;
 
+use MongoDB\BSON\Javascript;
```
Comment (jmikola) [inline src/Operation/MapReduce.php]: `finalize` should actually check for `JavascriptInterface`. I've opened [PHPLIB-313](https://jira.mongodb.org/browse/PHPLIB-313) to track that, and we'll fix this in 1.3.0.

File: src/ChangeStream.php (hunk 1 above)
Comment (jmikola) [reply src/ChangeStream.php]: Likely related to https://github.com/phpstan/phpstan/commit/679ea28f29699f409662cd81bd0414769b9e047e.

**Comment by jmikola** [comment]
Manually merged in https://github.com/mongodb/mongo-php-library/commit/8a73c759133332aee735ae6c37ad8b9f14fa1a62 with requested changes. Thanks!

**Comment by jmikola** [comment]
@carusogabriel: Side question: are there examples of incorporating PHPStan into `.travis.yml`? That may be something worth implementing so we can catch these issues more proactively going forward.

**Comment by carusogabriel** [comment]
@jmikola Sure. I'll open a PR with a new build running PHPStan, and we can discuss improvements there :+1:

**Pull request closed without merging**